collect_app/src/androidTest/java/org/odk/collect/android/LoadDefaultSharedPreferencesTest.java

```diff
@@ -0,0 +1,152 @@
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_HIGH_RESOLUTION;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_INSTANCE_SYNC;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_MAP_BASEMAP;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_MAP_SDK;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_NAVIGATION;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_PROTOCOL;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_SERVER_URL;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_SHOW_SPLASH;
+import static org.odk.collect.android.preferences.PreferenceKeys.KEY_SUBMISSION_URL;
+
+@RunWith(AndroidJUnit4.class)
+public class LoadDefaultSharedPreferencesTest {
+
+    @Before
+    public void setUp() {
+        GeneralSharedPreferences.getInstance().clear();
+        AdminSharedPreferences.getInstance().clear();
+    }
+
+    @Test
+    public void test() {
+        Map<String, ?> generalPreferences = GeneralSharedPreferences.getInstance().getAll();
+        Map<String, ?> adminPreferences = AdminSharedPreferences.getInstance().getAll();
+
+        assertTrue(generalPreferences.size() == 0);
+        assertTrue(adminPreferences.size() == 0);
+
+        Collect.getInstance().loadDefaultPreferences();
+        generalPreferences = GeneralSharedPreferences.getInstance().getAll();
+
+        // aggregate_preferences.xml
+        assertTrue(generalPreferences.containsKey(KEY_SERVER_URL));
+
+        // form_management_preferences.xml
+        assertTrue(generalPreferences.containsKey(KEY_AUTOSEND));
+        assertTrue(generalPreferences.containsKey(KEY_DELETE_AFTER_SEND));
+        assertTrue(generalPreferences.containsKey(KEY_COMPLETED_DEFAULT));
+        assertTrue(generalPreferences.containsKey(KEY_CONSTRAINT_BEHAVIOR));
+        assertTrue(generalPreferences.containsKey(KEY_HIGH_RESOLUTION));
+        assertTrue(generalPreferences.containsKey(KEY_INSTANCE_SYNC));
+
+        // identity_preferences.xml
+        assertTrue(generalPreferences.containsKey(KEY_ANALYTICS));
+
+        // other_preferences.xml
+        assertTrue(generalPreferences.containsKey(KEY_FORMLIST_URL));
+        assertTrue(generalPreferences.containsKey(KEY_SUBMISSION_URL));
+
+        // server_preferences.xml
+        assertTrue(generalPreferences.containsKey(KEY_PROTOCOL));
+
+        // user_interface_preferences.xml
+        assertTrue(generalPreferences.containsKey(KEY_FONT_SIZE));
+        assertTrue(generalPreferences.containsKey(KEY_NAVIGATION));
+        assertTrue(generalPreferences.containsKey(KEY_SHOW_SPLASH));
+        //assertTrue(generalPreferences.containsKey(KEY_SPLASH_PATH)); it should be in the collection too but it isn't probably because of the fact that PreferencesScreen can not have defaultValue? We should use Preferences instead of PreferencesScreen?
```

Comment:
There are a lot of strange things about this including that the default is localized and also used in tests. Would be worth revising how this setting works.

Reply:
We read this value here https://github.com/opendatakit/collect/blob/master/collect_app/src/main/java/org/odk/collect/android/activities/SplashScreenActivity.java#L91 and as I tested the expected default value is never recived it works because we add defValue in getString method.

Reply:
I'm pretty sure it's because we use PreferenceScreen https://github.com/opendatakit/collect/blob/master/collect_app/src/main/res/xml/user_interface_preferences.xml#L27 defaultValue for PreferenceScreen? doesn't sound good and and I think it's not allowed